src/core/processing/qgsprocessingexception.h

```diff
@@ -0,0 +1,35 @@
+/***************************************************************************
+                         qgsprocessingalgexception.h
+                         ------------------------
+    begin                : June 2017
+    copyright            : (C) 2017 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPROCESSINGEXCEPTION_H
+#define QGSPROCESSINGEXCEPTION_H
+
+#include "qgis_core.h"
+#include "qgsexception.h"
+
+/**
+ * \class QgsProcessingException
+ * \ingroup core
+ * Custom exception class for processing related exceptions.
+ */
+class CORE_EXPORT QgsProcessingException : public QgsException
```

Comment:
what about moving QgsException, QgsCsException and QgsProcessingException all to the same header?
they are almost the same and that would be coherent with the sip file.